### ISSUE TITLE
Separate model and data concerns on nodes and edges

### ISSUE THREAD

**Issue by jthrilly** [codaco/Network-Canvas#620]
Relevant for #446, #587, #596, #597  #615 

Currently we model nodes and edges as "flat" objects. This means that informational properties, such as values given to node/edge type variables), are at the same "level" as model properties, such as `_uid`.

This has caused several issues, including potential property collisions, the need to maintain lists of data variables for export, and reluctance to store useful meta data on nodes for fear of polluting the model further.

To address this I propose we introduce the following:

``` JavaScript
let node = {
  _private: {
    uid: '2343-2342-24gj-d2ff', //model data and metadata, including _uid etc.
  },
  ...properties //node variable values
}
```

### Assumptions
- Properties within `_private` will be documented for developers, but are not intended to be user facing. They will not be assumed to be exported as part of "the network", but might be used by Server and Architect in some way.
- Properties outside of `_private` should be the sole outcome of assigning a node/edge variable a value. All properties here should strictly correspond to variable registry variables. 
- This model will be implemented on both nodes and edges.

**Comment by jthrilly** [comment]
Other variables to be moved to `_private`:

stageId
promptId
createdIndex
timeCreated

**Comment by bryfox** [comment]
One way this encapsulation would be broken (with IDs in particular) is with user-provided worker scripts. If a user wanted to label a node based on membership in an edge, they would need to compare something like `edge.to === node._private.uid` ([akin to the current example](https://github.com/codaco/Network-Canvas/blob/4c6bd500d818a58802a852d770dfa67f75830fb9/public/protocols/development.netcanvas/nodeLabelWorker.js#L46)).

I suppose we could add some other object (`meta`?) based on the private contents and document that instead. Or maybe this is a use case for user-defined primary keys.

**Comment by bryfox** [comment]
To play devil's advocate: we should invert the described relationship, and encapsulate the model data.

```javascript
node: {
  uid: '2343-2342-24gj-d2ff',
  // ...other metadata...
  model: {
    ...properties //node variable values
  },
}
```

- no reserved words would be needed in keys (externalData, forms, etc.)
- uid may be the most frequently-accessed prop
- cleaner updates: [less care needed](https://github.com/codaco/Network-Canvas/blob/master/src/ducks/modules/network.js#L44-L55) to avoid mutating private state
- better encapsulation: we can freely pass `node`s around without worrying about omitting `_private`; 'private' vars are whitelisted, not blacklisted
  - Most of the issues listed in the description are an indirect result of not having clean access to the node's model data
  - (we could map redux state differently to react props, but having a single conceptual model would be cleaner)

**Comment by jthrilly** [comment]
I'm totally fine with the reversal. Seems like a reasonable thing to change. 

Can we do better than "model" for a name though? Is "properties" a protected word? It would make the most graph-theoretic sense.

**Comment by bryfox** [comment]
"properties" isn't reserved and sounds good to me.

**Comment by wwqrd** [comment]
Could relate to: https://github.com/codaco/Network-Canvas/issues/648

**Comment by jthrilly** [comment]
This has been implemented for nodes in #652. Needs implementation for edges.

**Comment by jthrilly** [comment]
Resolved for edges in https://github.com/codaco/Network-Canvas/pull/852
